pagopa/io-app
# fix(Firma con IO): [SFEQS-1271] Introducing URL polling to wait availability of filled document

## Short description
This PR adds an URL polling feature to wait availability of filled document pdf file. When the app request the QTSP clauses get also a `document_url`, an empty template mandatory for the QTSP that should be filled with the user data. The filling operations, made by the backend, it could be a long task operations so this polling mechanism fetch the pdf url using a `MAX_POLLING_RETRY` and a `POLLING_FREQ_TIMEOUT`.

To do this it has been defined a new state `FciPollFilledDocumentState`
```
{ isReady: boolean; retryTimes: number }
```
`isReady` is a boolean value (when true the filled document is ready [response === 200]) otherwise the saga try to fetch again the `filled_document_url` after `POLLING_FREQ_TIMEOUT` and `MAX_POLLING_RETRY` times.

## List of changes proposed in this pull request
- Add `fciPollFilledDocument` state
- Add new generator to manage url polling
- Update FCI saga
- Add reducers with some selectors
- Add test

## How to test
To test this improvement using the `io-dev-server-api` changing the `filled_document_url` in the FCI payloads with an invalid PDF url. Open reacttotron ans show the state sections to see `retryTimes` change on every request.
